src/Services/UserService.php

```diff
@@ -137,6 +137,69 @@ public function getUserByUUID($uuid)
         return $user;
     }
 
+    /**
+     * Retrieves a single user that is authorized for the calendar
+     * @param $userId
+     * @return array|null
+     */
+    public function getUserForCalendar($userId)
+    {
+        // TODO: eventually we'd like to leverage the inner search piece here and combine these methods
+        return $this->searchUsersForCalendar("", $userId);
+    }
+
+    /**
+     * Retrieves all the users that have been set to show up on the calendar optionally filtered by the facility if one
+     * is provided.
+     * @param string $facility
+     * @return array|null
+     */
+    public function getUsersForCalender($facility = "")
+    {
+        return $this->searchUsersForCalendar($facility);
+    }
+
+    private function searchUsersForCalendar($facility = "", $userId = null)
+    {
+        // this originally came from patient.inc::getProviderInfo()
+        $param1 = " AND authorized = 1 AND calendar = 1 ";
+        $bind = [];
+        if (!empty($userId)) {
+            $param1 .= " AND id = ? ";
+            $bind[] = $userId;
+        }
+
+        //--------------------------------
+        //(CHEMED) facility filter
+        $param2 = "";
+        if (!empty($facility)) {
+            if ($GLOBALS['restrict_user_facility']) {
+                // so is facility a string... or is it a column name... this is very confusing here
+                $param2 = " AND (facility_id = '" . add_escape_custom($facility) . "' OR  '" . add_escape_custom($facility)
+                    . "' IN (select facility_id from users_facility where tablename = 'users' and table_id = id))";
+            } else {
+                $param2 = " AND facility_id = '" . add_escape_custom($facility) . "' ";
```

Comment:
Could consider improving this part of the query by using binding (although not sure if the use of binding will work in that facility after the `OR`, although would be easy to test).

Reply:
I changed it to use binding and it appears to work fine with what I've tested.